src/Serilog/Context/LogContext.cs

```diff
@@ -12,9 +12,16 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
 #if LOGCONTEXT
```

Comment:
@merbla thanks for the refs. The idea is not to use either of those, but something like:

```csharp
[ThreadStatic]
static ImmutableStack<ILogEventEnricher> Data;
```

This should work on all platforms and does the job except for carrying state across async calls.